### ISSUE TITLE
Purpose of --assign-identity / description of the examples

### ISSUE THREAD

**Issue by BenjaminHerbert** [Azure/azure-cli#10359]
The examples mention different usages of the `--assign-identity` switch, e.g.:

&gt; Create a Linux multi-step task from a public GitHub repository with a git commit trigger and with the system-assigned managed identity

```
az acr task create -t hello-world:{{.Run.ID}} -n hello-world -r MyRegistry -c https://github.com/Azure-Samples/acr-tasks.git#:multipleRegistries -f testtask.yaml --assign-identity
```

Could you clarify how a task uses this identity and what this identity is used for? For me, it currently is unclear, also the information on https://docs.microsoft.com/en-us/cli/azure/acr/task/identity?view=azure-cli-latest does not help to clarify.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: bd0e3f3c-5aed-28b6-1e28-22ff32a8521d
* Version Independent ID: 128e5bc6-e518-bd5a-f511-48e0ed185739
* Content: [az acr task](https://docs.microsoft.com/en-us/cli/azure/acr/task?view=azure-cli-latest#az-acr-task-create)
* Content Source: [src/azure-cli/azure/cli/command_modules/acr/_help.py](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acr/_help.py)
* Service: **container-registry**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by jaysterp** [comment]
@BenjaminHerbert the `--assign-identity` flag is used to attach a managed identity to your ACR task. This managed identity can then be used to grant that task access to other Azure resources (ie. Azure Key Vault, another ACR). `--assign-identity` alone will provision a system-assigned identity for your task, while `--assign-identity <resourceID>` will assign an existing Azure Identity resource to your task. See these docs for further details on how to create a task with a managed identity and how to use this task to access secured resources:
[Enable a managed identity on a task](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-tasks-authentication-managed-identity)
[Cross-registry authentication from a task](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-tasks-cross-registry-authentication)
[External authentication using a key vault](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-tasks-authentication-key-vault)

**Comment by jaysterp** [comment]
@BenjaminHerbert also thank you for the feedback. We will improve our CLI reference docs to clarify how tasks use managed identity to avoid further confusion.

**Comment by northtyphoon** [comment]
Fixed by PR #10406
